packer/main.pkr.hcl

```diff
@@ -1,5 +1,73 @@
 packer {
-  required_version = ">= 1.7.0"
+  required_version = ">= 1.9.0"
+  required_plugins {
+    qemu = {
+      source  = "github.com/hashicorp/qemu"
+      version = "~> 1"
+    }
+    virtualbox = {
+      version = "~> 1"
+      source  = "github.com/hashicorp/virtualbox"
+    }
+  }
+}
+
+source "qemu" "kvm" {
+  cpus         = 2
+  memory       = 4096
+  disk_size    = 20480
+  machine_type = "virt"
+  accelerator  = "kvm"
+
+  format         = "qcow2"
+  headless       = "${var.headless}"
+  http_directory = "http"
+  qemu_binary    = "qemu-system-aarch64"
+  qemuargs = [
+    # Depending on the host OS this may also be /usr/share/qemu-efi/aarch64/QEMU_EFI.fd.
+    # Fedora & Arch both use the below path; Debian derivatives will use the above.
+    ["-bios", "/usr/share/edk2/aarch64/QEMU_EFI.fd" ],
+    ["-boot", "strict=off"],
+    ["-cpu", "host"],
+    ["-display", var.headless ? "none" : "gtk"],
+    ["-device", "virtio-rng-pci"],
+    ["-device", "virtio-gpu"],
+    ["-device", "nec-usb-xhci,id=xhci"],
+    ["-device", "usb-kbd,bus=xhci.0"],
+    ["-device", "usb-tablet,bus=xhci.0"]
+  ]
+  qemu_img_args {
+    create  = ["-o", "preallocation=falloc"]
+    convert = ["-o", "compression_type=zstd"]
+  }
+  disk_cache         = "unsafe"
+  disk_compression   = "true"
+  disk_detect_zeroes = "unmap"
+  disk_discard       = "unmap"
+  disk_interface     = "virtio"
+  net_device         = "virtio-net"
+  ssh_username       = "${var.ssh_user}"
+  ssh_password       = "${var.ssh_pass}"
+  ssh_timeout        = "100m"
+
+  # GRUB will wait 30s before choosing the default option, so 10s should
+  # give plenty of time to open the command line
+  boot_wait = "10s"
```

Comment:
RPi4 takes forever to even get to grub, and I've got this at 45sec in later commits. This will be a tricky one to support across widely different hardware.

Reply:
We can always make it a variable with some in-the-middle default (maybe 25s so we don't miss it on faster machines and maybe have a chance on slower ones?)